File: src/main/java/com/vinyl/model/Artist.java
package com.vinyl.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Artist {
	private String artistAlias;
	private Boolean isArtistActive;
	private String artistCountryCode;
	private String artistName;
	private Date artistBirthDate;
	private Date artistDeathDate;
	private List<String> currentBandAliases;
	private List<String> previousBandAliases;
	private List<String> trackCatalogNums;
	private List<String> featuringTrackCatalogNums;
}

File: src/main/resources/frontend/src/schemas/definitions/artist.js
export default {
  props: {
    artistAlias: 'string',
    isArtistActive: 'boolean',
    artistCountryCode: 'country',
    artistName: 'string',
    artistBirthDate: {
      type: 'date',
      typeConstraint: {
        to: []
      }
    },
    artistDeathDate: {
      type: 'date',
      typeConstraint: {
        to: []
      },
      isNullable: true
    },
    currentBandAliases: {
      type: 'band',
      isSchema: true,
      isArray: true,
      readonly: true,
      nofilter: true
    },
    previousBandAliases: {
      type: 'band',
      isSchema: true,
      isArray: true,
      readonly: true,
      nofilter: true
    },
    trackCatalogNums: {
      type: 'track',
      isSchema: true,
      isArray: true,
      readonly: true,
      nofilter: true
    },
    featuringTrackCatalogNums: {
      type: 'track',
      isSchema: true,
      isArray: true,
      readonly: true,
      nofilter: true
    },
    artistBindings: {
      type: 'artist2track',
      isSchema: true,
      isArray: true,
      filterOnly: true
    },
    participations: {
      type: 'participation',
      isSchema: true,
      isArray: true,
      filterOnly: true,
      joins: ['artist2band']
    }
  },
  key: 'artistAlias',
  display: (item) => item.artistAlias
}